src/blocks/PagePartialBlock/PagePartialBlock.astro

```diff
@@ -2,18 +2,26 @@
 import type { PagePartialBlockFragment } from '@lib/datocms/types';
 import GroupingBlock from '@blocks/GroupingBlock/GroupingBlock.astro';
 import type { GroupingBlockFragment } from '@lib/datocms/types';
+import { getEntry } from '@lib/content';
 
 export interface Props {
   block: PagePartialBlockFragment
 }
 const { block } = Astro.props;
 
+// TODO: only allow one page partial per block
+const pagePartials = await Promise.all(
+  block.items.map(async (item) => {
+    const pagePartial = await getEntry('PagePartials', item.id);
+    return pagePartial?.data;
+  })
+).then(items => items.filter(pagePartial => pagePartial !== undefined && pagePartial !== null));
```

Comment:
Ideally a PagePartialBlock only allows one PagePartial. Since that would be a big change, we're stil working on that, trying to find a way to write a migration that will move multiple pagePartials into a grouping block of page partials